hooks/command

```diff
@@ -1,575 +1,23 @@
-  workdir="${BUILDKITE_PLUGIN_DOCKER_WORKDIR:-$workdir_default}"
-fi
-
-# By default, mount $PWD onto $WORKDIR
-if [[ "${BUILDKITE_PLUGIN_DOCKER_MOUNT_CHECKOUT:-on}" =~ ^(true|on|1)$ ]] ; then
-  args+=( "--volume" "${pwd_default}:${workdir}" )
-fi
-
-# Parse volumes (and deprecated mounts) and add them to the docker args
-if plugin_read_list_into_result BUILDKITE_PLUGIN_DOCKER_VOLUMES BUILDKITE_PLUGIN_DOCKER_MOUNTS ; then
-  for arg in "${result[@]}" ; do
-    args+=( "--volume" "$(expand_relative_volume_path "${arg}")" )
-  done
-fi
-
-# If there's a git mirror, mount it so that git references can be followed.
-# But not if mount-checkout is disabled.
-if [[ -n "${BUILDKITE_REPO_MIRROR:-}" && "${BUILDKITE_PLUGIN_DOCKER_MOUNT_CHECKOUT:-on}" =~ ^(true|on|1)$ ]]; then
-  args+=( "--volume" "$BUILDKITE_REPO_MIRROR:$BUILDKITE_REPO_MIRROR:ro" )
-fi
-
-# Parse devices and add them to the docker args
-if plugin_read_list_into_result BUILDKITE_PLUGIN_DOCKER_DEVICES ; then
-  for arg in "${result[@]}" ; do
-    args+=( "--device" "${arg}" )
-  done
-fi
-
-# Parse sysctl args and add them to docker args
-if plugin_read_list_into_result BUILDKITE_PLUGIN_DOCKER_SYSCTLS ; then
-  for arg in "${result[@]}" ; do
-    args+=( "--sysctl" "$arg" )
-  done
-fi
-
-# Parse cap-add args and add them to docker args
-if plugin_read_list_into_result BUILDKITE_PLUGIN_DOCKER_ADD_CAPS ; then
-  for arg in "${result[@]}"; do
-    args+=("--cap-add" "$arg")
-  done
-fi
-
-# Parse cap-drop args and add them to docker args
-if plugin_read_list_into_result BUILDKITE_PLUGIN_DOCKER_DROP_CAPS ; then
-  for arg in "${result[@]}"; do
-    args+=("--cap-drop" "$arg")
-  done
-fi
-
-# Parse security-opts args and add them to docker args
-if plugin_read_list_into_result BUILDKITE_PLUGIN_DOCKER_SECURITY_OPTS ; then
-  for arg in "${result[@]}"; do
-    args+=("--security-opt" "$arg")
-  done
-fi
-
-# Parse ulimits args and add them to docker args
-if plugin_read_list_into_result BUILDKITE_PLUGIN_DOCKER_ULIMITS ; then
-  for arg in "${result[@]}"; do
-    args+=("--ulimit" "$arg")
-  done
-fi
-
-# Set workdir if one is provided or if the checkout is mounted
-if [[ -n "${workdir:-}" ]] || [[ "${BUILDKITE_PLUGIN_DOCKER_MOUNT_CHECKOUT:-on}" =~ ^(true|on|1)$ ]]; then
-  args+=("--workdir" "${workdir}")
-fi
-
-# Support docker run --user
-if [[ -n "${BUILDKITE_PLUGIN_DOCKER_USER:-}" ]] && [[ -n "${BUILDKITE_PLUGIN_DOCKER_PROPAGATE_UID_GID:-}" ]]; then
-  echo "+++ Error: Can't set both user and propagate-uid-gid"
-  exit 1
-fi
-
-if [[ -n "${BUILDKITE_PLUGIN_DOCKER_USER:-}" ]] ; then
-  args+=("-u" "${BUILDKITE_PLUGIN_DOCKER_USER:-}")
-fi
-
-# Parse publish args and add them to docker args
-if plugin_read_list_into_result BUILDKITE_PLUGIN_DOCKER_PUBLISH ; then
-  for arg in "${result[@]}" ; do
-    args+=( "--publish" "$arg" )
-  done
+if in_array "LOAD" "${commands[@]}" ; then
+  # shellcheck source=commands/load.sh
+  . "$DIR/../commands/load.sh"
```

Comment:
why source the script when you could just run it? (note that you will have to load the `shared.bash` library in the script as well if making the change)

Reply:
The reason for sourcing it instead of directly executing the script is just because I was keeping it similar to the structure of the docker-compose plugin where it's done the same way [here](https://github.com/buildkite-plugins/docker-compose-buildkite-plugin/blob/master/hooks/command#L31).

Reply:
Updated according to your comments, except this about sourcing because of the reason above. Let me know if you would rather have the script being ran instead of sourced and I'll change it.